tt_metal/distributed/mesh_device.cpp

```diff
@@ -100,8 +100,17 @@ SystemMesh& SystemMesh::instance() {
 }
 void SystemMesh::initialize() {
     this->physical_device_id_to_coordinate = tt::Cluster::instance().get_user_chip_ethernet_coordinates();
-    for (const auto& [chip_id, physical_coordinate] : this->physical_device_id_to_coordinate) {
-        this->physical_coordinate_to_device_id.emplace(physical_coordinate, chip_id);
+    if (this->physical_device_id_to_coordinate.empty()) {
+        // Only WH has ethernet coordinates. Fabric will assign chip ids for BH
+        for (auto chip_id = 0; chip_id < tt::umd::Cluster::detect_available_device_ids().size(); chip_id++) {
```

Comment:
This would call `detect_available_device_ids()` on every iteration of the loop, IIUC? Would caching the returned vector work, or perhaps this:
```
for (auto chip_id : tt::umd::Cluster::detect_available_device_ids()) {
```
This is technically a logical change, but I see `detect_available_device_ids` returns chip ids, and the variable name is the same...

Reply:
ah yeah, returns physical device ids. If for e.g. someone is allocated chip X out of N chips available on a machine it will give X rather than 0 even though they just run on one device. Since I need logical ids here I can uplift to pull out number of chips from cluster.

Reply:
Thanks for the context!